transcoders.go

```diff
@@ -121,61 +131,36 @@ func onionBtS(b []byte) (string, error) {
 	return addr + ":" + strconv.Itoa(int(port)), nil
 }
 
-var TranscoderP2P = NewTranscoderFromFunctions(p2pStB, p2pBtS)
+var TranscoderP2P = NewTranscoderFromFunctions(p2pStB, p2pBtS, p2pVal)
 
 func p2pStB(s string) ([]byte, error) {
 	// the address is a varint prefixed multihash string representation
 	m, err := mh.FromB58String(s)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse p2p addr: %s %s", s, err)
 	}
-	size := CodeToVarint(len(m))
-	b := append(size, m...)
-	return b, nil
+	return m, nil
 }
 
-func p2pBtS(b []byte) (string, error) {
-	// the address is a varint-prefixed multihash string representation
-	size, n, err := ReadVarintCode(b)
-	if err != nil {
-		return "", err
-	}
+func p2pVal(b []byte) error {
+	_, err := mh.Cast(b)
+	return err
+}
 
-	b = b[n:]
-	if len(b) != size {
-		return "", errors.New("inconsistent lengths")
-	}
+func p2pBtS(b []byte) (string, error) {
 	m, err := mh.Cast(b)
 	if err != nil {
 		return "", err
 	}
 	return m.B58String(), nil
 }
 
-var TranscoderUnix = NewTranscoderFromFunctions(unixStB, unixBtS)
+var TranscoderUnix = NewTranscoderFromFunctions(unixStB, unixBtS, nil)
 
 func unixStB(s string) ([]byte, error) {
-	// the address is the whole remaining string, prefixed by a varint len
-	size := CodeToVarint(len(s))
-	b := append(size, []byte(s)...)
-	return b, nil
+	return []byte(s), nil
 }
 
 func unixBtS(b []byte) (string, error) {
-	// the address is a varint len prefixed string
-	size, n, err := ReadVarintCode(b)
-	if err != nil {
-		return "", err
-	}
-
-	b = b[n:]
-	if len(b) != size {
-		return "", errors.New("inconsistent lengths")
-	}
-	if size == 0 {
-		return "", errors.New("invalid length")
-	}
-	s := string(b)
-	s = s[1:] // remove starting slash
-	return s, nil
+	return string(b), nil
```

Comment:
reasoning here?

Reply:
I wanted it to be symmetric (that is, the bytes to string function should be an inverse of the string to bytes function). I'm now handling this in `bytesToString` (as we do in `stringToBytes`).

Reply:
We could also not prepend a slash here:

https://github.com/multiformats/go-multiaddr/blob/84fb105635d33c95b41a9385064329b663f6f56f/codec.go#L43